### PR TITLE
clusterversion: mint 21.2 cluster version

### PR DESCRIPTION
**Pull request by celiala**
Shortly after [Creating a release branch](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/187859111/Creating+a+release+branch) for release-21.2, we'll want to merge PRs as instructed by the [New major version checklist](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/1522270228/New+major+version+checklist).

This PR is for Step 1a of the [New major version checklist](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/1522270228/New+major+version+checklist), where we add a cluster version `Start${vBRANCH_CUT}` for the corresponding `.0` release.

This PR is one of the tasks detailed in #70751, which tracks all the steps relevant to creating a release branch for `${vBRANCH_CUT}` and preparing master for the `${vNEXT}` major version.

Release justification: Non-production code change.
Release note: None

### CODE REVIEW

**Comment by celiala** [comment]
This should not be merged until after Tuesday's branch cut.

**Comment by celiala** [comment]
Tuesday's merge plan:

- https://github.com/cockroachdb/cockroach/pull/69826
- https://github.com/cockroachdb/cockroach/pull/69827
- https://github.com/cockroachdb/cockroach/pull/69828
- https://github.com/cockroachdb/cockroach/pull/69829
- (Plus backport of #69826 to release-21.2)

File: pkg/clusterversion/keyed_versions.go
Comment (celiala) [inline pkg/clusterversion/keyed_versions.go:93]: TODO(celia) - who might know the answer to this?

File: pkg/clusterversion/key_string.go
Comment (celiala) [inline pkg/clusterversion/key_string.go:2]: TODO(celia) - who might know the answer to this?

**Comment by dt** [comment]
I think we need to hold off on minting `21.2` until we have a release beta that is selected for our planned internal deployment -- once we mint the version, we cannot add new migrations/version gates under it without requiring a wipe of any clusters that ran a build including the mint.

**Comment by celiala** [comment]
> I think we need to hold off on minting 21.2 much longer than just the branch cut, and instead until we have a released beta that is selected for our planned internal deployment

Ack: I don't merge this until we've released a beta. Thanks!

I've updated the PR note and wiki runbook accordingly:

> **This should NOT be merged until we've released a beta ** (adding `do-not-merge` until this happens)

**Review by irfansharif — APPROVED** [review]
Code changes here LGTM (deferring to y'all for when this is to be merged).

**Review by knz — APPROVED** [review]
> Reviewed 4 of 7 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69826)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajwerner, @celiala, @dt, and @irfansharif)

___
*[pkg/clusterversion/key_string.go, line 2 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69826#-MihhIengxoZ6apJgmg5-r1-2:-MjE5kKuDCz9QSBcGhg1:bzeq6qb) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c84787f572843c93223d64c015d16f1d04d6ecc/pkg/clusterversion/key_string.go#L2)):*
<details><summary><i>Previously, celiala wrote…</i></summary><blockquote>

TODO(celia) - who might know the answer to this?
</blockquote></details>

answer to what?
___
*[pkg/clusterversion/keyed_versions.go, line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69826#-MihhIengxoZ6apJgmg6-r1-93:-MjE5xLA4KIgKCrMAKxD:b-eq3br6) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c84787f572843c93223d64c015d16f1d04d6ecc/pkg/clusterversion/keyed_versions.go#L93)):*
<details><summary><i>Previously, celiala wrote…</i></summary><blockquote>

TODO(celia) - who might know the answer to this?
</blockquote></details>

I believe this issue has been fixed. I think @dt or @irfansharif can confirm.

**Comment by celiala** [comment]
TFTR!

bors r+

**Comment by RaduBerinde** [comment]
Hi Celia,

CC is making some changes to the costing model and it requires a change to the tenant_usage table. Could you hold off 1/2 day on merging this change?

**Comment by celiala** [comment]
> CC is making some changes to the costing model and it requires a change to the tenant_usage table. Could you hold off 1/2 day on merging this change?

Ack. Will hold off until given green light to move forward, thanks! (re-adding `do-not-merge` label)

**Comment by RaduBerinde** [comment]
My PR (#70778) is merged. Sorry it took longer than predicted.

**Comment by celiala** [comment]
TFTRs!

bors r+

**Comment by celiala** [comment]
TFTRs!

bors r+

**Comment by celiala** [comment]
FYI - I'm now wondering aloud if I merged this too soon (this morning I just put 2+2 together around the context behind this change, as well as noticed one note that recommended we hold off under later -- perhaps until after an RC).

Because of that, I'll plan to hold off on backporting this until it seems like we have a pretty solid RC.

**Comment by dt** [comment]
waiting to backport until there's an RC tagged sounds about right to me. once we backport and release a 21.2 beta/rc with this change, that'll close the door on any future version gates or migrations being backported to 21.1, so we tend to wait until we get to an RC aka zero known release or GA blockers, just in case the resolution for any of those blockers ends up needing a gate.

**Pull request merged**
